src/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useRef, useState } from "react";
+import { OtpInputProps } from "./types";
+import {
+    CTAButton,
+    ErrorContainer,
+    InputContainer,
+    InputField,
+    Wrapper,
+} from "./otp-input.styles";
+
+export const OtpInput = ({
+    cooldownDuration,
+    ctaProps,
+    errorMessage,
+    numOfInput,
+    onChange,
+    ...otherProps
+}: OtpInputProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const {
+        disabled,
+        onClick,
+        styleType = "secondary",
+        ...otherCtaProps
+    } = ctaProps ?? {};
+    const inputRefs = useRef<(HTMLInputElement | null)[]>([]);
+
+    const [otpValues, setOTPValues] = useState<string[]>(
+        new Array(numOfInput).fill("")
+    );
+    const [countDown, setCountDown] = useState<number>(cooldownDuration);
+    const [enableCountDownTimer, setEnableCountDownTimer] =
```

Comment:
I am thinking if this flag might not be necessary. On the click of a button, we can just set the `countDown` to the `cooldownDuration`.